Builds failing with newer stdexec commits

E.g.
```
/opt/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.4.0/stdexec-git.8bc7c7f06fe39831dea6852407ebe7f6be8fa9fd_main-y6gargi7ilhiqdh6as5iudd75lhwzzl2/include/stdexec/__detail/__transform_sender.hpp:179:58: error: exception specification of 'decltype(auto) stdexec::dependent_domain::transform_sender(_Sender&&, const _Env&) const [with _Sender = stdexec::__sexpr<<lambda closure object><lambda()>(), stdexec::{anonymous}::__anon>; _Env = stdexec::__env::env<>]' depends on itself
```
on https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/7765210172/viewer.

GCC 11 stdexec pipelines will be disabled in https://github.com/eth-cscs/DLA-Future/pull/1180, and reenabled separately. Needs investigation if the issue is GCC 11 or the newer stdexec versions.

--- a/ci/.gitlab-ci.yml
+++ b/ci/.gitlab-ci.yml
@@ -5,9 +5,8 @@ include:
   - local: 'ci/cpu/clang15_release.yml'
   - local: 'ci/cpu/clang16_release.yml'
   - local: 'ci/cpu/clang18_release.yml'
-  # https://github.com/eth-cscs/DLA-Future/issues/1184
-  # - local: 'ci/cpu/gcc11_release_stdexec.yml'
-  # - local: 'ci/cpu/gcc11_debug_stdexec.yml'
+  - local: 'ci/cpu/gcc11_release_stdexec.yml'
+  - local: 'ci/cpu/gcc11_debug_stdexec.yml'
   - local: 'ci/cpu/gcc12_release_cxx20.yml'
   - local: 'ci/cpu/gcc13_codecov.yml'
   - local: 'ci/cpu/gcc13_release.yml'

--- a/include/dlaf/communication/kernels/internal/all_reduce.h
+++ b/include/dlaf/communication/kernels/internal/all_reduce.h
@@ -80,8 +80,9 @@ template <Device DCommIn, Device DCommOut, class T, Device DIn, Device DOut>
     // written into the output tile instead).  The reduction is explicitly done
     // on CPU memory so that we can manage potential asynchronous copies between
     // CPU and GPU. A reduction requires contiguous memory.
-    return withTemporaryTile<DCommIn, CopyToDestination::Yes, CopyFromDestination::No,
-                             RequireContiguous::Yes>(std::move(tile_in), std::move(all_reduce));
+    return pika::execution::experimental::make_unique_any_sender(
+        withTemporaryTile<DCommIn, CopyToDestination::Yes, CopyFromDestination::No,
+                          RequireContiguous::Yes>(std::move(tile_in), std::move(all_reduce)));
   };
 
 #if !defined(DLAF_WITH_MPI_GPU_AWARE)

